### PR TITLE
mana: Microsoft Azure Network Adapter (MANA) RDMA provider

### PR DESCRIPTION
**Pull request by longlimsft**
Introduce a provider that exposes MANA devices to user applications.

### CODE REVIEW

**Comment by longlimsft** [comment]
Hi,

Kernel patch is being reviewed at v4:
https://lore.kernel.org/all/1655345240-26411-1-git-send-email-longli@linuxonhyperv.com/T/#m50c39d647bc41f3641ca491b72b82aa225b81d66

Thanks,
Long

**Comment by jgunthorpe** [comment]
Please rebase it

The first commit must be generated with the kernel-headers/update script. Use commit ID 0266a177631d4c6b963b5b12dd986a8c5abdbf06

File: providers/mana/man/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+rdma_man_pages(
+  manadv.7.md
+  manadv_set_context_attr.3.md
+  manadv_init_obj.3.md
```
Comment (jgunthorpe) [inline providers/mana/man/CMakeLists.txt:4]: Keep sorted

**Review by jgunthorpe — CHANGES_REQUESTED** [review]

**Comment by rleon** [comment]
I don't see any progress here. It is not late to revert mana driver from upcoming PR to Linus.

**Comment by longlimsft** [comment]
Sorry, I was on vacation with limited access to lab.

I’m starting to work on this today.

________________________________

File: providers/mana/man/CMakeLists.txt (hunk 1 above)
Comment (longlimsft) [reply providers/mana/man/CMakeLists.txt:3]: Fixed, thank you.

**Comment by longlimsft** [comment]
> Please rebase it
> 
> The first commit must be generated with the kernel-headers/update script. Use commit ID 0266a177631d4c6b963b5b12dd986a8c5abdbf06

I have rebased the pull request. The first commit generated by kernel-headers/update script doesn't include an entry in rdma_kernel_provider_abi() in CMakeLists.txt, so I added the following with this commit:

@@ -70,6 +71,7 @@ rdma_kernel_provider_abi(
   rdma/hns-abi.h
   rdma/ib_user_verbs.h
   rdma/irdma-abi.h
+  rdma/mana-abi.h
   rdma/mlx4-abi.h
   rdma/mlx5-abi.h
   rdma/mthca-abi.h

Please let me know if this should be done in some other way.

**Pull request merged**